src/lib.rs

```diff
@@ -522,6 +541,30 @@ mod tests {
         }
     }
 
+    test_beat! {
```

Comment:
I think this behavior is wrong, this should probably be a separate test that asserts that running this is `Err(())`.